### PR TITLE
Add Bucket Locking option and fix versioning configuration.

### PR DESCRIPTION
**Pull request by dvaldivia**
We used to turn locking for any bucket with versioning features, not we only turn it on if there's retention configured or explicitly on, this means that we can have buckets with versioning enabled only

Screen Shot 2021-03-19 at 9 39 33 PM

### CODE REVIEW

File: portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // This file is part of MinIO Console Server
+// This file is part of MinIO Console Server
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket.tsx:2]: Please remove duplicated line

**Review by bexsoft — CHANGES_REQUESTED** [review]
Please fix the above, also it would be nice to show the object locking status in the bucket details screen
Screen Shot 2021-03-22 at 12 49 58

**Comment by belvid** [comment]
Hi!

The 3 lines below features should have a lighter gray color, since those are dividers and not actionable (as the input line above). I would suggest to use ##eaeaea for the lines as dividers (and apply the same to all similar cases)

File: portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket.tsx (hunk 1 above)
Comment (dvaldivia) [reply portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket.tsx:2]: what a weird accident

**Comment by bexsoft** [comment]
> Hi!
> 
> The 3 lines below features should have a lighter gray color, since those are dividers and not actionable (as the input line above). I would suggest to use ##eaeaea for the lines as dividers (and apply the same to all similar cases)

Sure! I will file a new ticket for it

**Comment by adfost** [comment]
Are the json changes necessary?

**Comment by bexsoft** [comment]
> Are the json changes necessary?

In this case, yes, as these files are auto generated by `make assets`

**Comment by dvaldivia** [comment]
@bexsoft done

Screen Shot 2021-03-22 at 1 03 29 PM

**Comment by dvaldivia** [comment]
Object Locking Enabled
Screen Shot 2021-03-22 at 1 43 09 PM


Object Locking Disabled

Screen Shot 2021-03-22 at 1 43 20 PM

**Comment by dvaldivia** [comment]
Screen Shot 2021-03-22 at 2 23 15 PM

Screen Shot 2021-03-22 at 2 23 20 PM

**Review by bexsoft — APPROVED** [review]
LGTM

**Review by Alevsk — APPROVED** [review]
LGTM

**Pull request merged**